ui/src/pages/namespace/Explorer/Page/poc/PageCompiler/primitives/Variable/LocalVariables.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  GlobalVariableScope,
+  Variable,
+  useGlobalVariableScope,
+} from "./VariableContext";
+
+import { LocalVariableNamespace } from "../../../schema/primitives/variable";
+
+type LocalVariableScope = Record<LocalVariableNamespace, Variable>;
+type VariableScope = GlobalVariableScope & LocalVariableScope;
+
+export type LocalVariables = VariableScope["this"];
+
+/**
+ * returns a function that allows to inject local
+ * variables into the global variables object.
+ */
+export const useLocalVariableInjection = () => {
+  const globalVariables = useGlobalVariableScope();
+  return (localVariables?: LocalVariables): VariableScope => ({
```

Comment:
```suggestion
/**
 * returns a function that accepts local variables, and returns a merged object
 * of context variables and local variables.
 */
export const useMergeLocalWithContextVariables = () => {
  const contextVariables = useVariablesContext();
  return (localVariables?: LocalVariables): VariableScope => ({
```
As discussed, "global" variables is misleading, and I think "context variables" sounds good, with the hook `useVariablesContext()`, respectively. (I would keep "variables" plural or it could be understood as an adjective!)

`useMergeLocalWithContextVariables` is a bit long, but accurate. `useMergeVariables()` would also work -- it doesn't explain everything, but I think the core idea is still preserved.